scripts/submit-analytics.js

```diff
@@ -0,0 +1,201 @@
+  const { machineId } = require("node-machine-id");
+
+  let identifier = await machineId();
+  if (!identifier) {
+    identifier = sha256("unknown");
+  }
+
+  let framework = "node.js";
+  let frameworkVersion = process.version;
+  let jsEngine = "v8";
+
+  if (context.dependencies && context.dependencies["react-native"]) {
+    framework = "react-native";
+    frameworkVersion = context.dependencies["react-native"];
+    try {
+      const podfile = fs.readFileSync("../../ios/Podfile", "utf8");
+      if (podfile.includes("hermes_enabled => true")) {
+        jsEngine = "hermes";
+      } else {
+        jsEngine = "jsc";
+      }
+    } catch (err) {
+      doLog(`Cannot read ios/Podfile: ${err}`);
+      jsEngine = "unknown";
+    }
+  }
+  if (context.dependencies && context.dependencies["electron"]) {
+    framework = "electron";
+    frameworkVersion = context.dependencies["electron"];
+  }
+
+  const payloads = {
+    webHook: {
+      event: "install",
+      properties: {
+        token: "aab85907a13e1ff44a95be539d9942a9",
+        "JS Analytics Version": 2,
+        distinct_id: identifier,
+        "Anonymized Machine Identifier": identifier,
+        "Anonymized Application ID": sha256(__dirname),
+        Binding: "javascript",
+        Version: context.version,
+        Language: "javascript",
+        Framework: framework,
+        "Framework Version": frameworkVersion,
+        "JavaScript Engine": jsEngine,
+        "Host OS Type": os.platform(),
+        "Host OS Version": os.release(),
+        "Node.js version": process.version,
+      },
+    },
+  };
+
+  return payloads;
+}
+
+function isAnalyticsDisabled() {
+  let isDisabled = false;
+
+  // NODE_ENV is commonly used by JavaScript framework
+  if ("NODE_ENV" in process.env) {
+    isDisabled |= process.env["NODE_ENV"] === "production" || process.env["NODE_ENV"] === "test";
+  }
+
+  // If the user has specifically opted-out or if we're running in a CI environment
+  isDisabled |= "REALM_DISABLE_ANALYTICS" in process.env || "CI" in process.env;
+
+  return isDisabled;
+}
+
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(payload) {
+  const https = require("https");
+
+  return new Promise((resolve, reject) => {
+    const webhookRequest = new Webhook().buildRequest(payload);
+
+    https
+      .get(webhookRequest, (res) => {
+        resolve({
+          statusCode: res.statusCode,
+          statusMessage: res.statusMessage,
+        });
+      })
+      .on("error", (error) => {
+        const message = error && error.message ? error.message : error;
+        const err = new Error(`Failed to dispatch analytics: ${message}`);
+        reject(err);
+      });
+  });
+}
+
+async function submitAnalytics(dryRun) {
+  if (isAnalyticsDisabled()) {
+    doLog("Analytics is disabled");
+    return;
+  }
+
+  const context = require("../../../package.json");
```

Comment:
If we added this to `@realm/react` then this would be one directory deeper.

Reply:
This could use the `npm_config_local_prefix` environment to get the path of the package we're getting installed into. We also have `fs-extra` as a dependency already, we might as well [use `readJsonSync` here](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/readJson-sync.md).

Also isn't it a biiiit excessive to send off the contents of the entire package.json? Isn't the `dependencies` object enough for what we want here?

Reply:
I dob't think we are sending off the entire `package.json` unless I misread the payload? https://github.com/realm/realm-js/blob/fe84d680184211c1adb602e28e821b58b6d09c1d/scripts/submit-analytics.js#L93

Reply:
Ahh - my bad. Read it too fast.

Reply:
I wonder if it is save to assume that we are interested in `package.json` in the same directory as `node_modules`. If so, we can use `process.cwd()` and chop off from `node_modules` and down.

Reply:
The working directory is set by NPM to our package root, so that won't work.